gdal-sys/src/lib.rs

```diff
@@ -4,4 +4,7 @@
 #![allow(clippy::upper_case_acronyms)]
 #![allow(rustdoc::bare_urls)]
 
+#[cfg(feature = "bundled")]
```

Comment:
That forces cargo to actually build and link that dependency. Otherwise it's clever and says that it is unused and therefore does not link the dependency.